samples/test-samples.cmake

```diff
@@ -95,9 +99,22 @@ endif ()
 # A variable to know if all samples ran successfully
```

Comment:
Can we add some check here to check the length of SAMPLES_LIST  equal to length of CRYPTO_LIB_LIST and generate fatal error if they don't match.

Reply:
Given that we manually set the `CRYPTO_LIB_LIST` along with `SAMPLES_LIST`, their length should be match.
If they do not match, either line 105 or 106 will fail eventually.